### PR TITLE
[merged] System containers mount scan

### PR DESCRIPTION
**Pull request by giuseppe**
system containers support for mount and scan commands

### CODE REVIEW

**Comment by rhatdan** [comment]
I think it would be better to keep system_container_exists() as a boolean, Or change it to get_system_container() and have it return None if it does not exist.  The function name makes no sense for a function that returns a tuple.

**Comment by rhatdan** [comment]
LGTM
@baude PTAL

**Comment by rhatdan** [comment]
Also branch needs update.

**Comment by giuseppe** [comment]
updated here :arrow_up:

File: Atomic/mount.py [hunk 1]
```diff
@@ -86,6 +87,10 @@ def set_args(self, args):
 
     def mount(self):
         try:
+            d = OSTreeMount(self.mountpoint, live=self.live, shared=self.shared)
+            if d.mount(self.image, self.live):
+                return
+
```
Comment (baude) [inline Atomic/mount.py]: Is this normally done where a variable assignment is used followed by a return and then used again ?

Comment (giuseppe) [inline Atomic/mount.py]: I have done it only to have a shorter line, since line 90 is already 80 columns

**Comment by rhatdan** [comment]
@giuseppe Please update your pull request and then we can merge.

**Comment by giuseppe** [comment]
rebased :arrow_up:

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
@cgwalters @jlebon Looks like something is wrong with jenkins?  Pip is blowing up?

**Comment by jlebon** [comment]
@rhatdan Hmmm seems like I have to add some toolchain pkgs for the pip package to compile

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by jlebon** [comment]
This should be fixed momentarily.

**Comment by jlebon** [comment]
The test machine is fixed, but some tests failed:

```
INTEGRATION TESTS:
Running test test_top.sh...		PASS
Running test test_printenv.sh...		PASS
Running test test_verify.sh...		FAIL
Running test test_dockertar_sha256_helper.sh...		PASS
Running test test_migrate.sh...		PASS
Running test test_mount.sh...		FAIL
Running test test_pass.sh...		PASS
Running test test_diff.sh...		PASS
Running test test_params.sh...		PASS
Running test test_display.sh...		PASS
Running test test_info.sh...		FAIL
Running test test_help.sh...		FAIL
```

I noticed some of them assume the existence of `/ostree/repo` ?

```
g-io-error-quark: /ostree/repo: open: No such file or directory (1)n+ [[ 0 != 1 ]]
```

@giuseppe Could these tests be changed so they set up their own ostree repo?

**Comment by giuseppe** [comment]
@jlebon, yes I am going to add a new patch to deal with it.

**Comment by jlebon** [comment]
Awesome!
@rh-atomic-bot retry

**Comment by cgwalters** [comment]
@rh-atomic-bot retry

**Comment by jlebon** [comment]
@rh-atomic-bot r+ 24d1cbb

File: test.sh [hunk 2]
```diff
@@ -11,6 +11,10 @@ export DOCKER=${DOCKER:-"/usr/bin/docker"}
 export SECRET=`dd if=/dev/urandom bs=4096 count=1 2> /dev/null | sha256sum`
 export ATOMIC_LIBEXEC="$(pwd)"
 
+mkdir $WORK_DIR/ostree-repo
+ostree --repo=$WORK_DIR/ostree-repo init --mode=archive-z2
```
Comment (cgwalters) [inline test.sh]: Shouldn't this be `--mode=bare-user` probably?  If we're using it for actual storage/checkouts.

Comment (giuseppe) [inline test.sh]: yes true.  Since the PR was already merged, I have a small fix here:

https://github.com/projectatomic/atomic/pull/388

**Pull request closed without merging**